### PR TITLE
feat (api): publish workflow run  notification

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: engine/api/workflow_queue.go [hunk 1]
```diff
@@ -351,7 +376,11 @@ func (api *API) postWorkflowJobStepStatusHandler() AsynchronousHandler {
 			return sdk.WrapError(err, "postWorkflowJobStepStatusHandler> Error while update job run")
 		}
 
-		return tx.Commit()
+		if err := tx.Commit(); err != nil {
+			return sdk.WrapError(err, "postWorkflowJobStepStatusHandler> Cannot ")
+		}
```
Comment (bnjjj) [inline engine/api/workflow_queue.go]: Cannot commit tx

File: sdk/event.go [hunk 2]
```diff
@@ -22,6 +22,48 @@ type EventEngine struct {
...
+	SubNumber       int64                     `json:"num,omitempty"`
+	Status          string                    `json:"status,omitempty"`
+	Start           int64                     `json:"start,omitempty"`
+	Done            int64                     `json:"done,omitempty"`
+	WorkflowName    string                    `json:"workflow_name,omitempty"`
+	PipelineName    string                    `json:"pipeline_name,omitempty"`
+	ProjectKey      string                    `json:"project_key,omitempty"`
+	ApplicationName string                    `json:"application_name,omitempty"`
+	EnvironmentName string                    `json:"environment_name,omitempty"`
+	Payload         interface{}               `json:"payload,omitempty"`
```
Comment (bnjjj) [inline sdk/event.go]: delete db

File: engine/api/workflow/workflow_run_event.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package workflow
+
+import (
```
Comment (yesnault) [inline engine/api/workflow/workflow_run_event.go]: import

File: engine/api/workflow/workflow_run_event.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/engine/api/event"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+)
+
+func GetWorkflowRunEventData(cError <-chan error, cEvent <-chan interface{}) ([]sdk.WorkflowRun, []sdk.WorkflowNodeRun, []sdk.WorkflowNodeJobRun, error) {
```
Comment (yesnault) [inline engine/api/workflow/workflow_run_event.go]: Doc

File: engine/api/workflow_queue.go [hunk 5]
```diff
@@ -13,7 +13,9 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/ovh/venom"
 
+	"github.com/go-gorp/gorp"
```
Comment (yesnault) [inline engine/api/workflow_queue.go]: import

File: engine/api/workflow_run.go [hunk 6]
```diff
@@ -10,7 +10,9 @@ import (
 
 	"github.com/gorilla/mux"
 
+	"github.com/go-gorp/gorp"
```
Comment (yesnault) [inline engine/api/workflow_run.go]: import

**Comment by fsamin** [comment]
I think a little think is missing here https://github.com/ovh/cds/blob/master/engine/api/repositoriesmanager/events.go#L58 to manage the new events to and send them to vcs

File: engine/api/workflow/dao_node_run_job.go [hunk 7]
```diff
@@ -147,12 +147,7 @@ func UpdateNodeJobRun(db gorp.SqlExecutor, store cache.Store, p *sdk.Project, j
 	if _, err := db.Update(&dbj); err != nil {
```
Comment (fsamin) [inline engine/api/workflow/dao_node_run_job.go]: `return  db.Update(&dbj)`

**Pull request merged**